src/Common/src/System/Security/Cryptography/RSAOpenSsl.cs

```diff
@@ -123,6 +120,35 @@ public override byte[] Decrypt(byte[] data, RSAEncryptionPadding padding)
             return plainBytes;
         }
 
+        public override bool TryDecrypt(ReadOnlySpan<byte> source, Span<byte> destination, RSAEncryptionPadding padding, out int bytesWritten)
+        {
+            if (padding == null)
+            {
+                throw new ArgumentNullException(nameof(padding));
+            }
+
+            Interop.Crypto.RsaPadding rsaPadding = GetInteropPadding(padding);
+            SafeRsaHandle key = _key.Value;
+            CheckInvalidKey(key);
+
+            if (destination.Length < Interop.Crypto.RsaSize(key))
```

Comment:
> have we tried using smaller-than-keysize values here

If we have a function that computes the required size based on all of those options you mention, and does so in a way that won't make OpenSSL blow up (keeping in mind that the native OpenSSL calls don't take a destination length and assume you've passed in something big enough) I'd be happy to use it.